util/progress/progressui/display.go

```diff
@@ -42,23 +52,32 @@ func WithDesc(text string, console string) DisplaySolveStatusOpt {
 	}
 }
 
-func DisplaySolveStatus(ctx context.Context, c console.Console, w io.Writer, ch chan *client.SolveStatus, opts ...DisplaySolveStatusOpt) ([]client.VertexWarning, error) {
-	modeConsole := c != nil
-
-	dsso := &displaySolveStatusOpts{}
-	for _, opt := range opts {
-		opt(dsso)
-	}
-
-	disp := &display{c: c, phase: dsso.phase, desc: dsso.consoleDesc}
-	printer := &textMux{w: w, desc: dsso.textDesc}
-
-	if disp.phase == "" {
-		disp.phase = "Building"
-	}
+type SolveStatusDisplay struct {
+	disp solveStatusDisplay
+}
 
-	t := newTrace(w, modeConsole)
+type solveStatusDisplay interface {
+	// init initializes the display and opens any resources
+	// that are required.
+	init(displayTimeout time.Duration)
+
+	// update sends the signal to update the display.
+	// Some displays will have buffered output and will not
+	// display changes for every status update.
+	update(ss *client.SolveStatus)
+
+	// refresh updates the display with the latest state.
+	// This method only does something with displays that
+	// have buffered output.
+	refresh()
```

Comment:
maybe `flush()` ?

Reply:
My intention behind this was to avoid coupling the intention with the implementation. I figured that the API for this was a bit similar to a double buffering API so `update` is used to update the internal state and then `refresh` is used to signal the display should be updated.

I can change it to `flush` but I felt the name `refresh` fit better for a display interface.